x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -167,20 +192,57 @@ public boolean equals(Object o) {
         QueryRuleCriteria that = (QueryRuleCriteria) o;
         return criteriaType == that.criteriaType
             && Objects.equals(criteriaMetadata, that.criteriaMetadata)
-            && Objects.equals(criteriaValue, that.criteriaValue);
+            && Objects.equals(criteriaValues, that.criteriaValues);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(criteriaType, criteriaMetadata, criteriaValue);
+        return Objects.hash(criteriaType, criteriaMetadata, criteriaValues);
     }
 
     @Override
     public String toString() {
         return Strings.toString(this);
     }
 
-    public boolean isMatch(String matchString) {
-        return criteriaType == CriteriaType.EXACT && criteriaValue.equals(matchString);
+    public boolean isMatch(Object matchValue, CriteriaType matchType) {
+        if (criteriaType == GLOBAL) {
+            return true;
+        }
+
+        final LevenshteinDistance ld = new LevenshteinDistance();
+        final String matchString = matchValue.toString();
+        boolean matchFound = false;
+
+        for (Object criteriaValue : criteriaValues) {
+            final String criteriaValueString = criteriaValue != null ? criteriaValue.toString() : null;
+            if (criteriaValueString != null && matchValue instanceof String) {
+                matchFound |= switch (matchType) {
+                    case EXACT -> matchString.equals(criteriaValueString);
+                    case EXACT_FUZZY -> ld.getDistance(matchString, criteriaValueString) > 0.75f; // TODO make configurable
+                    case PREFIX -> matchString.startsWith(criteriaValueString);
+                    case SUFFIX -> matchString.endsWith(criteriaValueString);
+                    case CONTAINS -> matchString.contains(criteriaValueString);
+                    default -> false;
+                };
+            } else if (criteriaValueString != null && matchValue instanceof Number) {
+                try {
+                    Number matchNumber = (Number) matchValue;
+                    matchFound |= switch (matchType) {
+                        case EXACT -> matchNumber.doubleValue() == Double.parseDouble(criteriaValueString);
```

Comment:
We can probably extract variables for `matchNumber.doubleValue()` and `Double.parseDouble(criteriaValueString)`